SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -621,6 +621,10 @@ theorem allOnes_shiftLeft_or_shiftLeft {x : BitVec w} (n : Nat) :
     BitVec.allOnes w <<< n ||| x <<< n = BitVec.allOnes w <<< n := by
   simp [← BitVec.shiftLeft_or_distrib]
 
+@[simp]
+theorem neg_eq_sub_zero (x : BitVec w) : - x = 0 - x := by
+    simp [bv_toNat]
```

Comment:
```suggestion
theorem zero_sub (x : BitVec w) : 0 - x = - x = := by
    simp [bv_toNat]
```

Reply:
We typically want the simper variant on the left. And we can name only the LHS if the RHS is obvious.

Reply:
Also for equalities where x arises on both sides, we can use implicit arguments.